src/user/models/user/user.service.ts

```diff
@@ -60,15 +64,27 @@ export class UserService {
     return await this.toDto(user, detailed);
   }
 
-  async getUserByAddress(address: string): Promise<User> {
-    return this.userRepo.findOne({ address });
+  async getUserByAddress(address: string, needsRelation = false): Promise<User> {
+    return this.userRepo.findOne({ where: { address }, relations: needsRelation ? ['userData', 'wallet'] : [] });
+  }
+
+  async getAllLinkedUsers(id: number): Promise<LinkedUserOutDto[]> {
+    return this.userRepo
+      .createQueryBuilder('user')
+      .select(
+        'linkedUser.address, linkedUser.blockchain, isSwitchable = CASE WHEN linkedUser.stakingBalance = 0 then CAST(1 AS BIT) else CAST(0 AS BIT) end',
+      )
+      .leftJoin('user.userData', 'userData')
+      .leftJoin('userData.users', 'linkedUser')
+      .where('user.id = :id', { id })
+      .getRawMany<LinkedUserOutDto>();
   }
 
   async getRefUser(ref: string): Promise<User> {
     return await this.userRepo.findOne({ where: { ref }, relations: ['userData'] });
   }
 
-  async createUser(dto: CreateUserDto, userIp: string, userOrigin?: string): Promise<User> {
+  async createUser(dto: CreateUserDto, userIp: string, userOrigin?: string, userData?: UserData): Promise<User> {
```

Comment:
Is the userData parameter used somewhere?

Reply:
no not right now, it is more of a preparation for users actively linking addresses. I have added it with the first link procedure and haven't removed it, as we will need it anyway again

Reply:
Ok, makes sense 👍